templates/build/news-feed-filters.build.hbs

```diff
@@ -14,8 +14,27 @@
         <div id="collapse-\{{id}}-\{{@index}}" class="panel-collapse collapse">
           <div class="panel-body">
             <fieldset data-filter-group>
-              \{{#each [data]}}
-              <div class="btn hidden-filter-controls-filter focus-outline" tabindex="0" data-toggle="\{{[class]}}" data-field="\{{../name}}" data-value="\{{[name]}}" >\{{[name]}}</div>
+              \{{#each  ../detailViewOptions}}
```

Comment:
Instead of using Handlebars to go through this logic (for which I feel makes it quite difficult to debug/update and enhance, I think the data going into the Handlebars template should be pre-processed so that Handlebars knows, for each filter, which type of filter it should render, e.g.

```js
var filters = [{ type: 'default', name: 'location' }, { type: 'date', name: 'DoB' }];
var tpl = Handlebars.compile('{{#each data}} {{#ifCond type '==' "date"}} <!-- Date filter markup --> {{else}} <!-- Standard filter markup --> {{/ifCond}} {{/each}}');
var html = tpl({ data: filters });
```

Reply:
Please advise, where it is possible to add type to the data set? Now we have only the name and class. And that's why I'm using the name of the column instead of type

Reply:
[This is the code](https://github.com/Fliplet/fliplet-widget-dynamic-lists/blob/0179acd511a85a05c6f19c3ead98a64674397b67/js/layout-javascript/news-feed-code.js#L1725-L1730) where the `filters` array (used for this template) is generated. I suggest passing the detail view options as a parameter to the `_this.Utils.Records.parseFilters()` function so that it can output the `filters` collection with a `type` property that's based on an assessment of the detail view options.